src/data/webData.ts

```diff
@@ -442,4 +443,231 @@ export function deleteSession(
             reject(error)
         }
     })
+}
+
+// ----- NEWS -----
+
+/**
+ * Converts a RawNews from the database into a News
+ * 
+ * @param rawNews
+ * @returns News
+ */
+function buildNews(rawNews: RawNews): News {
+    return {
+        id: rawNews.id,
+        title: rawNews.title,
+        subject: rawNews.subject,
+        body: rawNews.body,
+        postDate: new Date(rawNews.post_date),
+        imageURL: rawNews.image_url
+    } as News
+}
+
+/**
+ * Gets a single record from the News Table
+ * 
+ * @param id
+ * @returns News on found; null on not found
+ */
+function getNewsSync(
+    id: number
+): News | null {
+    let rawData = db.prepare(`
+    SELECT id, title, subject, body, post_date,, image_url FROM news 
+    WHERE id = ?
+    `)
+    .get(id) as RawNews | null
+
+    if (!rawData) return null
+    
+    return buildNews(rawData)
+}
+
+/**
+ * Gets a single record from the News Table
+ * 
+ * @param id
+ * @returns News on found; null on not found
+ */
+export function getNews(
+    id: number
+): Promise<News | null> {
+    return new Promise<News | null>((resolve, reject) => {
+        try {
+            resolve(getNewsSync(id))
+        } catch (error) {
+            reject(error)
+        }
+    })
+}
+
+/**
+ * Gets a News[] from the database based on startDate
+ * 
+ * @param startDate
+ * @param limit
+ * @param offset
+ * @returns News[] with all news for the given start date and offset
+ */
+function getNewsfeedSync(
+    startDate: Date,
+    limit: number,
+    offset: number
+): News[] {
+    let rawData = db.prepare(`
+    SELECT id, title, subject, body, post_date, image_url FROM news 
+    WHERE post_date > ? 
+    LIMIT ? OFFSET ?
+    `)
+    .all(startDate.toISOString(), limit, offset) as RawNews[]
+
+    return rawData.map(buildNews) as News[]
+}
```

Comment:
Can you make it so that all of the parameters can be nullable? I would give the limit a default value of 50, the offset a default value of 0, and the startDate just nullable.